samples/matter/lock/src/credentials/credentials_storage.cpp

```diff
@@ -35,14 +35,18 @@ bool GetStorageFreeSpace(size_t &freeBytes)
 
 /* Currently the secure storage is available only for non-Wi-Fi builds,
    because NCS Wi-Fi implementation does not support PSA API yet. */
-#ifdef CONFIG_CHIP_WIFI
+#if defined(CONFIG_NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND) &&                                                      \
+	!defined(CONFIG_NCS_SAMPLE_MATTER_SECURE_STORAGE_BACKEND)
 #define PSInit NonSecureInit
 #define PSStore NonSecureStore
 #define PSLoad NonSecureLoad
-#else
+#elif defined(CONFIG_NCS_SAMPLE_MATTER_SECURE_STORAGE_BACKEND) and                                                     \
+	!defined(CONFIG_NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND)
 #define PSInit SecureInit
 #define PSStore SecureStore
 #define PSLoad SecureLoad
+#else
+#error Please select only one of the possible credential backends: CONFIG_NCS_SAMPLE_MATTER_SECURE_STORAGE_BACKEND or CONFIG_NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND
```

Comment:
I'm not sure about this one. From the credential manager perspective this is true that we only need just one, but the application may use the other one for whatever reason, right?

Reply:
I've changed it to match your suggestion.